projects/plugins/jetpack/extensions/shared/block-editor-asset-loader.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Returns the current document and window contexts for `elementRef`.
+ * Use to retrieve the correct context for elements that may be within an iframe.
+ *
+ * @param   {HTMLElement} elementRef - The element whose context we want to return.
+ * @returns {Object}                 - The current document (`currentDoc`) and window (`currentWindow`) contexts.
+ */
+export function getLoadContext( elementRef ) {
+	const currentDoc = elementRef.ownerDocument;
+	const currentWindow = currentDoc.defaultView || currentDoc.parentWindow;
+
+	return { currentDoc, currentWindow };
+}
+
+/**
+ * Returns whether a given element is contained within an iframe.
+ * Useful to check if a block sits inside the Site Editor.
+ *
+ * @param   {HTMLElement} elementRef - The element whose context we want to return.
+ * @returns {boolean}                - Whether `elementRef` is contained within an iframe.
+ */
+export function isElementInIframe( elementRef ) {
+	const { currentWindow } = getLoadContext( elementRef );
+	return currentWindow.self !== currentWindow.top;
```

Comment:
We might need to add an additional check either here or within `getLoadContext` that we're in the _correct_ iframe (the site editor) and not within another version of an iframed editor (e.g. the post/page editor iframed within `wordpress.com/post/<blog_name>/<post_id>`).

Reply:
Oh wow, thanks for double checking that. 🤕 

I did check in WPCOM when I was initially targeting `'iframe[name="editor-canvas"]'` 

I'll try to reinstate that test. 

THANK YOU

Reply:
I re-added an iframe name check. Not ideal, but the whole thing is (theoretically) transitory.

I could try to check in the back end and add a global JS variable if it might help future endeavours.

Reply:
We could also add to test that the iframe passes the same-origin-policy before we try to start injecting things into the header and bork it all again. :D